src/DynamoCore/Graph/Workspaces/HomeWorkspaceModel.cs

```diff
@@ -744,15 +744,19 @@ private void OnUpdateGraphCompleted(AsyncTask task)
             }
 
             var workspace = updateTask.TargetedWorkspace;
-            foreach (var message in messages)
-            {
-                var guid = message.Key;
-                var node = workspace.Nodes.FirstOrDefault(n => n.GUID == guid);
-                if (node == null)
-                    continue;
+            //wrap all calls that will result in dispatch calls into a single call.
+            DispatchOnUIThread(new Action(()=>{
+                foreach (var message in messages)
+                {
+                    var guid = message.Key;
+                    var node = workspace.Nodes.FirstOrDefault(n => n.GUID == guid);
+                    if (node == null)
+                        continue;
 
-                node.Warning(message.Value); // Update node warning message.
-            }
+                    node.Warning(message.Value); // Update node warning message.
+                }
+            }),false);
+           
```

Comment:
so before this code...the `node.Warning(message.Value)` call was invoked in the context of the Scheduler thread ?

Reply:
yes, exactly, and foreach node foreach message update that triggers a dispatch to the UI thread.

Reply:
I don;t see any locking mechanisms in the warning logic...
I wonder if there could be issues with that..